### PR TITLE
Sequential fractional

### PR DESCRIPTION
**Pull request by merowin**
Add 2 player variant with fractional stones and sequential moves.

issue: #254

### CODE REVIEW

File: packages/vue-client/src/components/GameCreation/SFractionalConfigForm.vue [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+function emitConfigChange() {
+  emit("configChanged", config);
+}
+
+function setBoardConfig(boardConfig: BoardConfig): void {
+  config.board = boardConfig;
+  emitConfigChange();
+}
+
+/* TODO: Figure out how to make this work with vues reactivity
```
Comment (benjaminpjones) [inline packages/vue-client/src/components/GameCreation/SFractionalConfigForm.vue:28]: I think you just need to wrap `config` in `reactive()`:

```
const config: SFractionalConfig = reactive({
  ...props.initialConfig,
  board: { ...props.initialConfig.board },
  secondary_colors: [...props.initialConfig.secondary_colors],
});
```

Vue is capable of detecting updates on reactive arrays!

**Review by benjaminpjones — COMMENTED** [review]
It's a cool variant!  Generlly, I think it would be better to scope the changes only to what is related to sfractional, but otherwise I think this will be a nice addition!

File: packages/vue-client/src/components/GameCreation/ParallelGoConfigForm.vue [hunk 2]
```diff
@@ -30,7 +30,7 @@ function emitConfigChange() {
   </form>
 </template>
 
-<style>
+<style scoped>
```
[the remark this answers is not in the document]
Comment (merowin) [reply packages/vue-client/src/components/GameCreation/ParallelGoConfigForm.vue:33]: I agree, I got a bit carried away, sorry. I was trying to make the input fields align at the right, and discovered that the unscoped styles were applied.
I'll undo these changes, and make a separate pr for the style changes.

**Review by benjaminpjones — APPROVED** [review]
Thanks for the clean up! I think 14 files is more reasonable (esp. considering vue changes are mostly optional for variant implementers)

**Comment by merowin** [comment]
Hm there is some weird behaviour where clicking the + button in the creation form triggers a page reload. I'll need to find out the reason.

File: packages/vue-client/src/assets/main.css [hunk 3]
```diff
@@ -113,4 +113,4 @@ a,
 .info-label {
   font-weight: bold;
   margin-right: 0.5em;
-}
\ No newline at end of file
+}
```
Comment (JonKo314) [inline packages/vue-client/src/assets/main.css:116]: Why delete the empty line at the end of that file? Was that some auto formatting?

File: packages/vue-client/src/components/boards/SFractional/SFractionalGridBoard.vue
Comment (JonKo314) [inline packages/vue-client/src/components/boards/SFractional/SFractionalGridBoard.vue:1]: `SFractionalGraphBoard.vue` and SFractionalGridBoard.vue` are pretty similar, I wonder if they should be put into one file.

Comment (merowin) [reply packages/vue-client/src/components/boards/SFractional/SFractionalGridBoard.vue:1]: I think the code is easier to read and cleaner with separate files in this case.

File: packages/vue-client/src/assets/main.css (hunk 3 above)
Comment (merowin) [reply packages/vue-client/src/assets/main.css:116]: I don't know to be honest. Should I revert it?

**Comment by merowin** [comment]
> Hm there is some weird behaviour where clicking the + button in the creation form triggers a page reload. I'll need to find out the reason.

Fixed, seems like buttons in forms trigger a page reload. Maybe like a submit form button would.

File: packages/vue-client/src/components/GameCreation/SFractionalConfigForm.vue [hunk 4]
```diff
@@ -45,30 +45,30 @@ function removeSecondaryColor(): void {
...
+    <label>Secondary Colors</label>
+
     <input
       type="color"
       v-for="(_, index) in config.secondary_colors"
       :key="index"
       v-model="config.secondary_colors[index]"
     />
   </form>
+  <!-- Note: I would like to place these buttons above the color inputs rather than below. -->
```
Comment (benjaminpjones) [inline packages/vue-client/src/components/GameCreation/SFractionalConfigForm.vue:58]: Possible fix: `type="button"`

https://stackoverflow.com/questions/7803814/how-can-i-prevent-refresh-of-page-when-button-inside-form-is-clicked

File: packages/vue-client/src/assets/main.css (hunk 3 above)
Comment (JonKo314) [reply packages/vue-client/src/assets/main.css]: Maybe I misunderstood. I think GitHub is saying the `}` without `\n` was removed and a `}\n` was added, but the newline isn't visible in GitHub. Anyway, if the auto format thinks it should be this way, there's no need to revert it.

File: packages/vue-client/src/components/boards/SFractional/SFractionalGridBoard.vue
Comment (JonKo314) [reply packages/vue-client/src/components/boards/SFractional/SFractionalGridBoard.vue:1]: Yes, it's easier to read. But as it is also a lot of duplicated code, changes in the future will probably need to be done twice. It's not a big deal though.

**Review by JonKo314 — APPROVED** [review]

**Comment by merowin** [comment]
Thank you both for the review!

**Pull request merged**